MIGRATION_GUIDE.md

```diff
@@ -367,6 +367,14 @@ are all internal to the driver.
 This should have 0 impact as they were internal utilities unreachable from
 the public APIs.
 
+### Private functions in interfaces
+
+Both `neo4j.SessionWithContext` and `neo4j.ResultWithContext` define unexported functions.
+These are needed to avoid duplication between the `context.Context`-aware interface and their legacy, pre v5, variants.
+This may however cause issues with code generation-based mocking libraries.
```

Comment:
```suggestion
This may, however, cause issues with code-generation-based mocking libraries.
```
Or no hyphens? Or `[...] based on code generation`? :thinking: